### PR TITLE
Update error code in Create and Update Exp.

### LINKED ISSUE #567: Specific http status code if record already exist in DB
API Response for POST/PUT call should contain specify response code if record already exist in the database.

### PR DESCRIPTION
**Pull request by khansaad**
This PR fixes #567

### CODE REVIEW

File: src/main/java/com/autotune/analyzer/experiment/ExperimentResultValidation.java [hunk 1]
```diff
@@ -88,29 +89,28 @@ public void validate(List<ExperimentResultData> experimentResultDataList, Map<St
                                     proceed = true;
                             } else {
                                 proceed = false;
-                                resultData.setValidationResultData(new ValidationOutputData(false, errorMsg));
+                                resultData.setValidationResultData(new ValidationOutputData(false, errorMsg, HttpServletResponse.SC_BAD_REQUEST));
```
Comment (dinogun) [inline src/main/java/com/autotune/analyzer/experiment/ExperimentResultValidation.java:92]: Looks like this method call should have been `setValidationOutputData`, got missed out in #659

Comment (khansaad) [reply src/main/java/com/autotune/analyzer/experiment/ExperimentResultValidation.java]: Updated wherever missed

File: design/CreateExperiment.md [hunk 2]
```diff
@@ -29,18 +29,27 @@ tuned.
   * Duplicate Experiment name.
   ```
   {
-  "message": "Experiment name already exsist with Kruize. View registered experiments at /listExperiments",
-  "httpcode": 400,
+  "message": "Experiment name already exists",
```
Comment (dinogun) [inline design/CreateExperiment.md:32]: Please format the json

File: design/PerformanceProfile.md [hunk 3]
```diff
@@ -61,19 +61,28 @@ tuned.
...
   "httpcode": 400,
   "documentationLink": "",
-  "status": "FAILED"
+  "status": "ERROR"
+  }
+  ```
+  * Server Error
+  ```
+  {
+  "message": "Internal Server Error",
```
Comment (dinogun) [inline design/PerformanceProfile.md:82]: When is this error being returned?

File: design/UpdateResults.md [hunk 4]
```diff
@@ -1,3 +1,66 @@
...
+  }
+  ```
+    * Mandatory attribute value are missing.
+  ```
+  {
+  "message": "experiment_name and timestamp are mandatory fields.",
+  "httpcode": 400,
+  "documentationLink": "",
+  "status": "FAILED"
+  }
```
Comment (dinogun) [inline design/UpdateResults.md:56]: Also update the `MonitoringModeAPI.md` doc to mention the parameters that are mandatory. ie `cpuUsage`, `memoryUsage` and `memoryRSS`.
Also mention that if the parameters `cpuRequest`, `cpuLimit`, `memoryRequest` and `memoryLimit` are missing, then they are assumed to not have been set for the container in question.

Comment (khansaad) [reply design/UpdateResults.md:56]: Added

File: design/UpdateResults.md [hunk 5]
```diff
@@ -1,3 +1,66 @@
...
+```
+
+* Failure
+    * Experiment name not found.
+  ```
+  {
+  "message": "Experiment name: <experiment-name> not found",
+  "httpcode": 400,
+  "documentationLink": "",
+  "status": "FAILED"
```
[the remark this answers is not in the document]
Comment (khansaad) [reply design/UpdateResults.md]: Done

File: design/PerformanceProfile.md (hunk 3 above)
Comment (khansaad) [reply design/PerformanceProfile.md]: Updated to "Any unknown exception on server side"

File: design/CreateExperiment.md [hunk 6]
```diff
@@ -29,18 +29,27 @@ tuned.
...
+  "httpcode": 409,
   "documentationLink": "",
   "status": "FAILED"
   }
   ```
   * Mandatory attribute value are missing.
```
[the remark this answers is not in the document]
Comment (khansaad) [reply design/CreateExperiment.md]: Updated

File: design/CreateExperiment.md (hunk 2 above)
Comment (khansaad) [reply design/CreateExperiment.md]: Done

**Pull request merged**